src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -26,26 +26,155 @@ hydrostatic pressure) to the current model state. If `callbacks` are provided (i
+    for region in 1:6
+
+      if mod(region,2) == 1
+        #- odd face number (1,3,5):
+        region_E = mod(region + 0, 6) + 1
+        region_N = mod(region + 1, 6) + 1
+        region_W = mod(region + 3, 6) + 1
+        region_S = mod(region + 4, 6) + 1
+        for k in -Hz+1:Nz+Hz
+        #- N + S Halo for field₁:
+            field₁[region][iMn:iMx, Nc+1:Nc+Hc, k] .= reverse(field₂[region_N][1:Hc, iMn:iMx, k],dims=2)'*plmn
+            field₁[region][iMn:iMx, 1-Hc:0, k] .=             field₁[region_S][iMn:iMx, Nc+1-Hc:Nc, k]
+        #- E + W Halo for field₂:
+            field₂[region][Nc+1:Nc+Hc, iMn:iMx, k] .=     field₂[region_E][1:Hc, iMn:iMx, k]
+            field₂[region][1-Hc:0, iMn:iMx, k] .= reverse(field₁[region_W][iMn:iMx, Nc+1-Hc:Nc, k],dims=1)'*plmn
+        end
+      else
+        #- even face number (2,4,6):
+        region_E = mod(region + 1, 6) + 1
+        region_N = mod(region + 0, 6) + 1
+        region_W = mod(region + 4, 6) + 1
+        region_S = mod(region + 3, 6) + 1
+        for k in -Hz+1:Nz+Hz
+        #- N + S Halo for field₁:
+            field₁[region][iMn:iMx, Nc+1:Nc+Hc, k] .=     field₁[region_N][iMn:iMx, 1:Hc, k]
+            field₁[region][iMn:iMx, 1-Hc:0, k] .= reverse(field₂[region_S][Nc+1-Hc:Nc, iMn:iMx, k],dims=2)'*plmn
+        #- E + W Halo for field₂:
+            field₂[region][Nc+1:Nc+Hc, iMn:iMx, k] .= reverse(field₁[region_E][iMn:iMx, 1:Hc, k],dims=1)'*plmn
+            field₂[region][1-Hc:0, iMn:iMx, k] .=             field₂[region_W][Nc+1-Hc:Nc, iMn:iMx, k]
+        end
+      end
+
+    end
+
+    #-- Add one valid field₁, field₂ value next to the corner, that allows
+    #   to compute vorticity on a wider stencil (e.g., vort3(0,1) & (1,0))
+    for region in 1:6
+        #println("region=",region,", size(field₁)=",size(field₁[region][:,:,1]),
+        #                         ", size(field₂)=",size(field₂[region][:,:,1]) )
+        for k in -Hz+1:Nz+Hz
+            #- SW corner:
+            field₁[region][1-Hc:0, 0, k] .= field₂[region][1, 1-Hc:0, k]
+            field₂[region][0, 1-Hc:0, k] .= field₁[region][1-Hc:0, 1, k]'
+        end
+        if Hc > 1
+          for k in -Hz+1:Nz+Hz
+            #- NW corner:
+            field₁[region][2-Hc:0, Nc+1, k] .= reverse(field₂[region][1, Nc+2:Nc+Hc, k])*plmn
+            field₂[region][0, Nc+2:Nc+Hc, k] .= reverse(field₁[region][2-Hc:0,  Nc,  k])'*plmn
+            #- SE corner:
+            field₁[region][Nc+2:Nc+Hc, 0, k] .= reverse(field₂[region][ Nc,  2-Hc:0, k])*plmn
+            field₂[region][Nc+1, 2-Hc:0, k] .= reverse(field₁[region][Nc+2:Nc+Hc, 1, k])'*plmn
+            #- NE corner:
+            field₁[region][Nc+2:Nc+Hc, Nc+1, k] .= field₂[region][Nc, Nc+2:Nc+Hc, k]
+            field₂[region][Nc+1, Nc+2:Nc+Hc, k] .= field₁[region][Nc+2:Nc+Hc, Nc, k]'
+          end
+        end
+    end
+
+    return nothing
+end
 
 function update_state!(model::HydrostaticFreeSurfaceModel, grid, callbacks; compute_tendencies = true)
```

Comment:
why do we need to modify `update_state!` to fill the halo regions for the grid coords/metrics?

Reply:
You are right. We don't. The modification results from this script being copied from the ncc-glw/cubed-sphere-dynamics branch associated with PR 3306, where we have decided to employ the same function fill_paired_halo_regions! to fill the halos of the grid metrics (with signed=false) and the velocities (with signed=true). Anyway, I am reverting this modification for the purpose of this PR in a subsequent commit.